src/srcset.js

```diff
@@ -210,16 +210,19 @@ export class Srcset {
         // minimum values seen.
         sourceWidth = minWidth / 2;
       }
-      minWidth = Math.min(minWidth, sourceWidth);
-      // The calculation is slightly biased toward higher width by offsetting
-      // score by negative 0.2.
-      const score = Math.abs((sourceWidth - width) / width - 0.2);
-      if (score < minScore) {
-        minScore = score;
-        minIndex = i;
+      // First candidate width that's equal or higher than the requested width
+      // will stop the search.
+      if (sourceWidth >= width) {
+        // The right value is now between `i` and `i + 1` - select the one
+        // that is closer with a slight preference toward higher numbers.
+        const delta = sourceWidth - width;
+        const prevDelta = width - prevWidth * 0.9;
```

Comment:
I understand the comment, but the implementation doesn't follow that:

```js
const prevWidth = 1900;
const width = 1920;
const sourceWidth = 2000;

//...
const delta = 2000 - 1920; // => 80
const prevDelta = 1920 - 1900 * .9; // => 210
```

That `prevDelta` is drastic, even though it's the better choice. Now take:

```js
const prevWidth = 1919;
const width = 1920;
const sourceWidth = 2000;

//...
const delta = 2000 - 1920; // => 80
const prevDelta = 1920 - 1919 * .9; // => 192.9
```

Even 1px difference makes us swing to the larger value every time. Instead, we could do:

```js
const prevDelta = (width - prevWidth) * 1.1;
(1920 - 1900) * 1.1; // => 22
(1920 - 1919) * 1.1; // => 1.1

// The "turning" point
(1920 - 1847) * 1.1; // => 80.3
```

Now we get proper 10% penalties.